panic: decimal division by 0 in liquidity_provision.go:169 ...

... or thereabouts: 

The artifacts are here: https://jenkins.vega.rocks/job/common/job/vega-market-sim-reinforcement/23/artifact/network_home/vega-sim-qbk5mxn_/ 

including hopefully a replay log. 

```
panic: decimal division by 0

goroutine 424742 [running]:
github.com/shopspring/decimal.Decimal.QuoRem({0xc0296310e0, 0x0}, {0xc029631120, 0x0}, 0x10)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:590 +0x2d8
github.com/shopspring/decimal.Decimal.DivRound({0xc0296310e0?, 0x51743c8?}, {0xc029631120?, 0x40?}, 0x10)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:632 +0x59
github.com/shopspring/decimal.Decimal.Div(...)
	/jenkins/GOPATH/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:577
code.vegaprotocol.io/vega/core/execution/common.(*MarketLiquidity).applyPendingProvisions(0xc00872b860, {0x4020478, 0xc0072c6ba0}, {0x0?, 0xc009e62f00?, 0x0?}, 0xc00ddc3660)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/common/liquidity_provision.go:169 +0x726
code.vegaprotocol.io/vega/core/execution/common.(*MarketLiquidity).OnEpochStart(0xc00872b860, {0x4020478, 0xc0072c6ba0}, {0x0?, 0x0?, 0x0?}, 0xc002d4fbc0?, 0xc002d620d0?, 0xc002d620d0?, {0xc00997c180, ...})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/common/liquidity_provision.go:297 +0xaa
code.vegaprotocol.io/vega/core/execution/future.(*Market).OnEpochEvent(0xc00997e500, {0x4020478, 0xc0072c6ba0}, {0x5b, {0x0, 0xedbec7ace, 0x0}, {0x0, 0xedbec7bfa, 0x0}, ...})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/future/market.go:348 +0x17c
code.vegaprotocol.io/vega/core/execution.(*Engine).OnEpochEvent(...)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/execution/engine.go:235
code.vegaprotocol.io/vega/core/epochtime.(*Svc).notify(0xc00157c120, {0x4020478, 0xc0072c6ba0}, {0x5b, {0x0, 0xedbec7ace, 0x0}, {0x0, 0xedbec7bfa, 0x0}, ...})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/epochtime/service.go:201 +0xe5
code.vegaprotocol.io/vega/core/epochtime.(*Svc).OnTick(0xc00157c120, {0x4020478, 0xc0072c6ba0}, {0x1?, 0x403aaf0?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/epochtime/service.go:142 +0x854
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/vegatime/service.go:104
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0xc00050c380, {0x4020478, 0xc0072c6ba0}, {0x40?, 0x302aa00?, 0x0?})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/vegatime/service.go:77 +0x24f
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(_, {{0xc00af50b00, 0x20, 0x20}, {{0x0, 0x0}, {0xc00032f1c0, 0x6}, 0x6a2f, {0x0, ...}, ...}, ...})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/processor/abci.go:825 +0x747
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).BeginBlock(_, {{0xc00af50b00, 0x20, 0x20}, {{0x0, 0x0}, {0xc00032f1c0, 0x6}, 0x6a2f, {0x0, ...}, ...}, ...})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/abci/abci.go:46 +0x75
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).BeginBlock(_, {{0xc00af50b00, 0x20, 0x20}, {{0x0, 0x0}, {0xc00032f1c0, 0x6}, 0x6a2f, {0x0, ...}, ...}, ...})
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/cmd/vega/commands/node/app_wrapper.go:64 +0x55
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).BeginBlock(0xc000236240)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:336 +0x399
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0xc000236240)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:195 +0x24b
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0xc000236240, 0x60db88400)
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/nullchain.go:272 +0x1e5
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:129 +0x2a
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/jenkins/workspace/common/vega-market-sim-reinforcement/extern/vega/core/blockchain/nullchain/server.go:128 +0x24a
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -177,6 +177,7 @@
 - [9305](https://github.com/vegaprotocol/vega/issues/9305) - Properly propagate `providersFeeCalculationTimeStep` parameter
 - [9374](https://github.com/vegaprotocol/vega/issues/9374) - `ListGovernanceData` ignored reference filter.
 - [9385](https://github.com/vegaprotocol/vega/issues/9385) - Support deprecated liquidity auction type for compatibility
+- [9398](https://github.com/vegaprotocol/vega/issues/9398) - Fix division by zero panic in market liquidity
 
 ## 0.72.1
 

--- a/core/execution/common/liquidity_provision.go
+++ b/core/execution/common/liquidity_provision.go
@@ -154,6 +154,10 @@ func (m *MarketLiquidity) applyPendingProvisions(
 
 		// min(-proposedCommitmentVariation, bondAccountBalance)
 		commitmentVariation := num.MinD(proposedCommitmentVariation.Neg(), acc.Balance.ToDecimal())
+		if commitmentVariation.IsZero() {
+			continue
+		}
+
 		commitmentVariationPerParty[partyID] = commitmentVariation
 		sumOfCommitmentVariations = sumOfCommitmentVariations.Add(commitmentVariation)
 	}

